src/mon/NVMeofGwMap.cc

```diff
@@ -217,12 +218,16 @@ int NVMeofGwMap::do_delete_gw(
 int NVMeofGwMap::get_num_namespaces(const NvmeGwId &gw_id,
     const NvmeGroupKey& group_key,  const BeaconSubsystems&  subs)
 {
-  auto grpid = created_gws[group_key][gw_id].ana_grp_id ;
+  auto grpid = created_gws[group_key][gw_id].ana_grp_id;
   int num_ns = 0;
-  for (auto & subs_it:subs) {
-    for (auto & ns :subs_it.namespaces) {
-      if (ns.anagrpid == (grpid+1)) {
-         num_ns++;
+  if (subs.size() == 0) {
+    dout(4) << "Empty subsystems  for GW " << gw_id  << dendl;
+  } else {
+    for (auto & subs_it:subs) {
```

Comment:
The type of subs_it is not an iterator -- don't suffix it with _it.

It the idea here simply to add debug output for the empty subs case?  dout(4) is likely too high for this case.  It's not really all that useful to skip the loop for empty subs, mostly just makes the code harder to parse.